src/meta/client/MetaClient.cpp

```diff
@@ -39,8 +41,27 @@ MetaClient::~MetaClient() {
 }
 
 
-void MetaClient::init() {
+bool MetaClient::isMetadReady() {
+    if (sendHeartBeat_) {
+        auto ret = heartbeat().get();
+        if (!ret.ok()) {
+            LOG(ERROR) << "Heartbeat failed, status:" << ret.status();
+            ready_ = false;
+            return ready_;
+        }
+    }  // end if
     loadDataThreadFunc();
+    return ready_;
+}
+
+
+bool MetaClient::waitForMetadReady(int count, int retryIntervalSecs) {
```

Comment:
On what condition does this function fail?

Reply:
Next version, we will add clusterId support, when mete client meets wrong clusterId, then will fail.
This version, it will be all success!